testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1184,6 +1193,22 @@ public void testRenameColumn()
         super.testRenameColumn();
     }
 
+    @Test
+    public void testInsertNegativeDate()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_INSERT));
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "insert_date", "(dt DATE)")) {
+            if (hasBehavior(SUPPORTS_NEGATIVE_DATE)) {
+                assertUpdate(format("INSERT INTO %s VALUES (DATE '-2016-12-07')", table.getName()), 1);
+                assertQuery("SELECT * FROM " + table.getName(), "VALUES DATE '-2016-12-07'");
+            }
+            else {
+                assertQueryFails(format("INSERT INTO %s VALUES (DATE '-2016-12-07')", table.getName()), "(?s).*Failed to insert data.*");
```

Comment:
-2016 is quite distant past.
let's use more modern negative date, like -1 

(especially for negative test)